Mathlib/LinearAlgebra/TensorProduct/Tower.lean

```diff
@@ -0,0 +1,205 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison, Johan Commelin
+-/
+import Mathlib.LinearAlgebra.TensorProduct
+import Mathlib.Algebra.Algebra.Tower
+
+#align_import ring_theory.tensor_product from "leanprover-community/mathlib"@"88fcdc3da43943f5b01925deddaa5bf0c0e85e4e"
+
+/-!
+# The `A`-module structure on `M ⊗[R] N`
+
+When `M` is both an `R`-module and an `A`-module, and `Algebra R A`, then many of the morphisms
+preserve the actions by `A`.
+
+This file provides more general versions of the definitions already in
+`LinearAlgebra/TensorProduct`.
+
+## Main definitions
+
+ * `TensorProduct.AlgebraTensorModule.curry`
+ * `TensorProduct.AlgebraTensorModule.uncurry`
+ * `TensorProduct.AlgebraTensorModule.lcurry`
+ * `TensorProduct.AlgebraTensorModule.lift`
+ * `TensorProduct.AlgebraTensorModule.lift.equiv`
+ * `TensorProduct.AlgebraTensorModule.mk`
+ * `TensorProduct.AlgebraTensorModule.assoc`
+
+## Implementation notes
+
+We could thus consider replacing the less general definitions with these ones. If we do this, we
+probably should still implement the less general ones as abbreviations to the more general ones with
+fewer type arguments.
```

Comment:
This note is just copied from the other file: it's maybe still a good idea, but I didn't want to attempt it yet as it interferes with some other refactors I have in progress.